store/router.go

```diff
@@ -11,6 +13,14 @@ import (
 	"github.com/ethereum/go-ethereum/log"
 )
 
+type IRouter interface {
```

Comment:
nice abstraction! Are you thinking of adding a different implementation of Router? like cachingRouter/fallbackRouter that specifically add caching/fallback?

Reply:
hmm that could be nice to have but fail to the see current use cases or necessity of decoupling here

Reply:
dependent on how need shifts in the future it could be valuable though